discovery/spec.md

```diff
@@ -365,775 +244,693 @@ The following sections define the attributes that appear in a Service entity.
     }
     ````
 
-##### specversions
+#### channel
 
-- Type: Array of `String` values
-- Description: CloudEvents
-  [`specversions`](../cloudevents/spec.md#specversion)
-  that can be used for events published for this service.
+- Type: String
+- Description: A string that can be used to correlate Endpoints. Any Endpoints
+  within an instance of a Discovery Service that share the same `channel` value
+  MUST have some relationship. This specification does not define that
+  relationship or the specific values used in this property. 
+
+  This property MAY be serialized as an empty string or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
-  - REQUIRED
-  - MUST be a non-empty array of non-empty strings
+  - OPTIONAL
+  - if present, MUST be a string
+- Examples:
+  - `queue1`
+
+#### authscope
+
+- Type: TBD
+- Description: TBD
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - TBD
 
-##### subscriptionurl
+#### usage
 
-- Type: `URL`
-- Description: An absolute URL indicating where CloudSubscriptions `subscribe`
-  API calls MUST be sent to.
+- Type: String
+- Description: The interaction model supported by this Endpoint. This
+  specification defines a set of possible values (see the
+  [Usage Values](#usage-values) section for more information, however,
+  implementations MAY define new values as extensions. It is RECOMMENDED
+  that extension values be defined with some organizational unique string
+  to reduce the chances of collisions with possible future spec-defined values.
 - Constraints:
   - REQUIRED
+  - MUST be a non-empty string
+- Examples:
+  - `consumer`
+
+#### config
 
-##### subscriptionconfig
+- Type: Object
+- Description: Metadata the provides additional information concerning the
+  interactions with this Endpoint. Each `usage` value MUST define the set of
+  `config` properties that are valid.
 
-- Type: `Map` of `String` to `String`
-- Description: A map indicating supported options for the `config` parameter for
-  the CloudSubscriptions subscribe() API call. Keys are the name of keys in the
-  allowed config map, the values indicate the type of that parameter, conforming
-  to the CloudEvents
-  [type system](../cloudevents/spec.md#type-system).
-  TODO: Needs resolution with CloudSubscriptions API
+  This property MAY be serialized as an empty object or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
   - OPTIONAL
 - Examples:
-  - ??
+  - `{ "protocol": "kafka" , "endpoint": "https://example.com/queue/1234" }`
+
+#### groups
 
-##### subscriptiondialects
+- Type: Array
+- Description: A set of Definition Groups supported by this Endpoint. Each
+  Group MAY be a full representation of the Group or MAY be a
```

Comment:
Personally, I worry having the either or here for full definition or ref (via which, `id` or `uri`??) makes the implementation complicated for both the consumer and the discovery server here.

I would prefer a `$ref` style pointer and have it be a URI-Ref

Reply:
Do you mean to not allow in-lining at all? We wanted to be able to support the "get it all in one GET" scenario, which is what in-lining does. However, we could mandate that for the "single GET" scenario people need to always query from the top of the resource model since that allows for many different resource types to be returned - then a local ref could always be used. This would mean that a GET on an individual resource group would always have refs. Would this break anyone's model/usecases?